### PR TITLE
Remove 'TSGForOI' module

### PR DESCRIPTION
**Pull request by andrea21z**
#### PR description:

This PR is created to remove a module that are not used anymore in production:
`TSGForOI` from `RecoMuon/TrackerSeedGenerator`.

- The issue is described in [#37973](https://github.com/cms-sw/cmssw/issues/37973).
- The deprecated module was replaced by `TSGForOIFromL2` in 2018, and this one was replaced in April by `TSGForOIDNN` (`RecoMuon/TrackerSeedGenerator/plugins/TSGForOIDNN.h`).
- Files related with this module have been removed.
- We update the L3MuonProducer [test](https://github.com/cms-sw/cmssw/blob/master/RecoMuon/L3MuonProducer/test/newL3.py) to run `TSGForOIDNN` instead of `TSGForOI`.

#### PR validation:

The proposed changes passed all code format and runTheMatrix tests.

### CODE REVIEW

File: RecoMuon/L3MuonProducer/test/newL3.py [hunk 1]
```diff
@@ -202,31 +202,26 @@ def producers_by_type(process, type):
...
-                SF3 = cms.double(5.0),
-                SF4 = cms.double(7.0),
-                SF5 = cms.double(10.0),
-		hitsToTry = cms.int32(5),
-	        layersToTry = cms.int32(3),
-	        maxEtaForTOB = cms.double(1.1),
-	        maxSeeds = cms.uint32(1),
-	        minEtaForTEC = cms.double(0.9),
-	        src = cms.InputTag("hltL2Muons","UpdatedAtVtx")
+    process.hltIterL3OISeedsFromL2Muons = cms.EDProducer("TSGForOIDNN",
```
Comment (missirol) [inline RecoMuon/L3MuonProducer/test/newL3.py:205]: This indentation is incorrect. The file is actually broken in the first place (try `python3 newL3.py`), b/c of mixed use of tabs and whitespaces. Please fix it by removing tabs, and reformatting (or, remove the file altogether if the MUO POG prefers that).

**Comment by missirol** [comment]
assign muon-pog

See question in https://github.com/cms-sw/cmssw/pull/39726#issuecomment-1277801148.

File: RecoMuon/L3MuonProducer/test/newL3.py (hunk 1 above)
Comment (andrea21z) [reply RecoMuon/L3MuonProducer/test/newL3.py]: I have fixed the indentation problem, now it should run correctly. For the tabs/whitespaces part, I will wait to see if we want to remove the file or keep it.

**Comment by mandrenguyen** [comment]
please test

**Comment by missirol** [comment]
please abort

Let's save resources.

The file in test/ still needs to be either fixed or removed based on the preference of the MUO POG. That file is not tested in PR tests (well, it is not tested anywhere..), and it is the only thing that changed wrt previous tests.

**Comment by missirol** [comment]
@cms-sw/muon-pog-l2 

Could you please review this PR?

**Comment by khaosmos93** [comment]
Hi @missirol, we confirm that it can be safely removed, sorry for the delayed answer.

**Comment by missirol** [comment]
Thanks, @khaosmos93 . There is a pending item, https://github.com/cms-sw/cmssw/pull/39726#discussion_r995631175. The `newL3.py` file is currently broken (`python3 newL3.py` does not work). I suggest to fix it, unless you prefer to remove it. Please let @andrea21z know your feedback.

File: RecoMuon/L3MuonProducer/test/newL3.py (hunk 1 above)
Comment (khaosmos93) [reply RecoMuon/L3MuonProducer/test/newL3.py]: Hi @andrea21z, this customizer is obsolete and does not compatible with the Run 3 menu. So I suggest you remove the file.

Comment (andrea21z) [reply RecoMuon/L3MuonProducer/test/newL3.py]: I agree, I will remove the test file.

**Comment by missirol** [comment]
unassign muon-pog

I take https://github.com/cms-sw/cmssw/pull/39726#issuecomment-1280773412 as a sign-off of this PR by the MUO POG.

**Comment by missirol** [comment]
+hlt

Since the only difference is the removal of a file that wasn't used anywhere, the upcoming tests should pass just like they did before.

**Comment by mandrenguyen** [comment]
+1

**Comment by perrotta** [comment]
+1

**Pull request merged**